freshmaker/pyxis_gql.py

```diff
@@ -21,37 +21,22 @@
 
 from functools import cached_property
 
-from gql import gql, Client
+from gql import Client, gql
 from gql.dsl import DSLQuery, DSLSchema, dsl_gql
 from gql.transport.requests import RequestsHTTPTransport
+from requests_kerberos import OPTIONAL, HTTPKerberosAuth
 
 
 class PyxisGQLRequestError(RuntimeError):
     pass
 
 
-class RequestsHTTPTransportWithCert(RequestsHTTPTransport):
-    """A modified requests transport to support certificate authentication"""
-
-    def __init__(self, *args, **kwargs):
-        self.cert = kwargs.pop("cert", None)
-        if self.cert is None:
-            raise RuntimeError("Missing required keyword argument: cert")
-        super().__init__(*args, **kwargs)
-
-    def connect(self):
-        super().connect()
-        self.session.cert = self.cert
-
-
 class PyxisGQL:
-    def __init__(self, url, cert):
+    def __init__(self, url):
         """Create authenticated Pyxis GraphQL session"""
-        transport = RequestsHTTPTransportWithCert(
-            url=url,
-            retries=3,
-            cert=cert,
-        )
+        pyxis_krb_auth = HTTPKerberosAuth(mutual_authentication=OPTIONAL, force_preemptive=True)
+        transport = RequestsHTTPTransport(url=url, retries=3, auth=pyxis_krb_auth, verify=False)
```

Comment:
```suggestion
        transport = RequestsHTTPTransport(url=url, retries=3, auth=pyxis_krb_auth)
```
I don't remember why `verify=False` was added before, let's remove it.

Reply:
I believe this flag turns off the SSL verification during the requests. If I don't set it to false, I run into the following error when I try to make any of the queries:

> `requests.exceptions.SSLError: HTTPSConnectionPool(host='graphql.pyxis.dev.engineering.redhat.com', port=443): Max retries exceeded with url: /graphql/ (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)'))) `

I believe this flag will be necessary. Is it okay if we keep it?

Reply:
You need to have the internal Root CA installed, if you're running inside a virtual env and the `python-requests`/`python-certifi` are installed with `pip`, then you have to workaround this by:
```
$ export REQUESTS_CA_BUNDLE=/etc/pki/tls/certs/ca-bundle.crt
```
or
```
$ cp /etc/pki/tls/certs/ca-bundle.crt $(python -c "from requests.certs import where; print(where())")
```
I'll share the doc about installing the internal CA in jira, when you're not sure whether it's safe or not, it's better to not expose any internal links (like the pyxis graphql server url).

Reply:
That was exactly what I was missing, thanks a lot!

I removed the 'verify=False' flag and tested everything again just to be sure. I'll squash this together into this commit.